modules/standard/CTypes.chpl

```diff
@@ -836,16 +836,16 @@ module CTypes {
   */
   @chpldoc.nodoc
   inline proc c_ptrTo(ref arr: []): c_ptr(arr.eltType) {
-    if (!arr.isRectangular() || !arr.domain.distribution._value.dsiIsLayout()) then
+    if (!isSubtype(arr.domain._instance.type, DefaultRectangularDom)) then
       compilerError("Only single-locale rectangular arrays support c_ptrTo() at present");
 
-    if (arr._value.locale != here) then
-      halt(
-          "c_ptrTo() can only be applied to an array from the locale on " +
-          "which it lives (array is on locale " + arr._value.locale.id:string +
-          ", call was made on locale " + here.id:string + ")");
-
     if boundsChecking {
+      if (arr._value.locale != here) then
+        halt(
```

Comment:
Technically this might manifest as a change in program behavior from the user's point of view. Before they'd get a halt when compiling with ``--fast``, and with this change it would probably lead to a segfault.

Reply:
I pinged the team on Slack. You're right that it is a behavior change. I personally think that that's beyond our stability guarantees.

Reply:
It looks like we're OK with the change.